### PR TITLE
Start extracting common WARNING_CFLAGS into CommonBase.xcconfig

### PR DESCRIPTION
**Pull request by ddkilzer**
#### 2a6683c96f6be3ce8e348678b776b9b89844547a
<pre>
Start extracting common WARNING_CFLAGS into CommonBase.xcconfig
<a href="https://bugs.webkit.org/show_bug.cgi?id=250431">https://bugs.webkit.org/show_bug.cgi?id=250431</a>
&lt;rdar://104102992&gt;

Reviewed by Alex Christensen.

Extract common WARNING_CFLAGS into CommonBase.xcconfig into the
WK_COMMON_WARNING_CFLAGS Xcode variable.  The following warnings
are also removed:
  -W: Same as -Wextra.
  -Wmissing-format-attribute: No effect in clang (gcc compat).

Some projects also add WK_FIXME_WARNING_CFLAGS to document and
to disable pre-existing warnings that weren&apos;t previously known
because some warning flags were not enabled on some projects.

* Configurations/CommonBase.xcconfig:
(WK_COMMON_WARNING_CFLAGS): Add.
- Start list of common warning flags among all projects.
* Source/JavaScriptCore/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
(ANGLE (Dynamic)):
- Remove `GCC_TREAT_WARNINGS_AS_ERRORS = NO` from Debug, Release
  and Production configurations.  Regressed in 244928@main.
* Source/ThirdParty/ANGLE/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Source/ThirdParty/gtest/xcode/Config/General.xcconfig:
(GCC_TREAT_WARNINGS_AS_ERRORS): Add.  Use instead of -Werror.
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Source/WTF/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
* Source/WebCore/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
* Source/WebCore/PAL/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
* Source/WebCore/PAL/ThirdParty/libavif/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
* Source/WebCore/PAL/ThirdParty/libavif/ThirdParty/dav1d/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
* Source/WebGPU/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
* Source/WebInspectorUI/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Source/WebKit/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Source/WebKitLegacy/mac/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Source/bmalloc/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Tools/ImageDiff/cg/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Tools/MiniBrowser/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Tools/MobileMiniBrowser/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Tools/WebEditingTester/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Tools/WebKitTestRunner/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.
* Tools/lldb/lldbWebKitTester/Configurations/Base.xcconfig:
(WARNING_CFLAGS):
(WK_FIXME_WARNING_CFLAGS): Add.

Canonical link: <a href="https://commits.webkit.org/258855@main">https://commits.webkit.org/258855@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/f1446793f7c2ff204ae7908dd6fbacbbd1546931

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102911 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12030 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35934 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112162 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/172378 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106872 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13047 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2936 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95152 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109820 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108685 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10023 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93219 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37656 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/91867 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24751 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79392 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/93105 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5474 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26165 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/89480 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/3169 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5625 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2615 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/29689 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11637 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45658 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/92407 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6086 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7373 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/20692 "Passed tests") | 
| | | | |

### CODE REVIEW

**Comment by aproskuryakov** [comment]
I'm a bit torn on this, as:
- We certainly want to deduplicate these, and to fix any mistakes that accumulated by omission.
- But having the same warnings on our code, 3rd party code used in the product, and on things like gtest seems potentially inappropriate. It could be a lot of work to make 3rd party projects warning free when we want to enable more common warnings. 

What is the approach that balances these? You probably have something reasonable implemented in the code, but it's not explained in the commit message.

File: Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig [hunk 1]
```diff
@@ -72,7 +72,11 @@ GCC_WARN_UNUSED_FUNCTION = YES;
 GCC_WARN_UNUSED_VARIABLE = YES;
 CLANG_WARN_SUSPICIOUS_MOVE = YES;
 PREBINDING = NO;
-WARNING_CFLAGS = -Wexit-time-destructors -Wformat=2 -Wglobal-constructors -Wthread-safety;
+WARNING_CFLAGS = $(WK_COMMON_WARNING_CFLAGS) $(WK_FIXME_WARNING_CFLAGS) -Wexit-time-destructors -Wglobal-constructors -Wthread-safety;
+
+// Remove WK_FIXME_WARNING_CFLAGS once all warnings are fixed.
+// -Wno-unknown-warning-option added for -Wno-unused-but-set-parameter.
+WK_FIXME_WARNING_CFLAGS = -Wno-conditional-uninitialized -Wno-deprecated-copy -Wno-missing-field-initializers -Wno-sign-compare -Wno-undef -Wno-unknown-warning-option -Wno-unused-but-set-parameter -Wno-unused-but-set-variable -Wno-unused-parameter;
```
Comment (achristensen07) [inline Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:79]: I have a feeling these will never be fixed.

**Review by achristensen07 — APPROVED** [review]

**Comment by ddkilzer** [comment]
> What is the approach that balances these? You probably have something reasonable implemented in the code, but it's not explained in the commit message.

It's the `WK_FIXME_WARNING_CFLAGS` variable which collects the unfixed warnings for that project.

This is the language I added to the commit message:

```
    Some projects also add WK_FIXME_WARNING_CFLAGS to document and
    to disable pre-existing warnings that weren't previously known
    because some warning flags were not enabled on some projects.
```

Do you have specific suggestions for how to improve this description?

**Comment by aproskuryakov** [comment]
This addition looks good to me, thank you.

File: Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig (hunk 1 above)
Comment (ddkilzer) [reply Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:79]: > I have a feeling these will never be fixed.

1. I feel it's better to document them than to leave them as unknowns, even if they weren't fixed.
2. I literally plan to fix some (not all) of these after this patch lands as many have fewer than 10 results, but this patch was already getting big, so I didn't want to wrap up all those changes into this commit.

**Pull request merged**